contrib/babelfishpg_tsql/src/collation.c

```diff
@@ -274,6 +279,329 @@ transform_funcexpr(Node *node)
+
+	/* extract pattern */
+	pstatus = pattern_fixed_prefix_wrapper(patt, 1, server_collation_oid,
+											&prefix, NULL);
+
+	/* If there is no constant prefix then there's nothing more to do */
+	if (pstatus == Pattern_Prefix_None)
+	{
+		return node;
+	}
+
+	/*
+	 * If we found an exact-match pattern, generate an "=" indexqual.
+	 */
+	if (pstatus == Pattern_Prefix_Exact)
+	{
+		op_str = like_entry.is_not_match ? "<>" : "=";
+		optup = compatible_oper(NULL, list_make1(makeString(op_str)), ltypeId, ltypeId,
+								true, -1);
+		if (optup == (Operator) NULL)
+			return node;
+
+		ret = (Node *) (make_op_with_func(oprid(optup), BOOLOID, false,
+											(Expr *) leftop, (Expr *) prefix,
+											InvalidOid, server_collation_oid, oprfuncid(optup)));
+
+		ReleaseSysCache(optup);
+	}
+	else
+	{
+		Expr	   *greater_equal,
+					*less_equal,
+					*concat_expr;
+		Node	   *constant_suffix;
+		Const	   *highest_sort_key;
+
+		/* construct leftop >= pattern */
+		optup = compatible_oper(NULL, list_make1(makeString(">=")), ltypeId, ltypeId,
+								true, -1);
+		if (optup == (Operator) NULL)
+			return node;
+		greater_equal = make_op_with_func(oprid(optup), BOOLOID, false,
+											(Expr *) leftop, (Expr *) prefix,
+											InvalidOid, server_collation_oid, oprfuncid(optup));
+		ReleaseSysCache(optup);
+		/* construct pattern||E'\uFFFF' */
+		highest_sort_key = makeConst(TEXTOID, -1, server_collation_oid, -1,
+										PointerGetDatum(cstring_to_text(SORT_KEY_STR)), false, false);
+
+		optup = compatible_oper(NULL, list_make1(makeString("||")), rtypeId, rtypeId,
+								true, -1);
+		if (optup == (Operator) NULL)
+			return node;
+		concat_expr = make_op_with_func(oprid(optup), rtypeId, false,
+										(Expr *) prefix, (Expr *) highest_sort_key,
+										InvalidOid, server_collation_oid, oprfuncid(optup));
+		ReleaseSysCache(optup);
+		/* construct leftop < pattern */
+		optup = compatible_oper(NULL, list_make1(makeString("<")), ltypeId, ltypeId,
+								true, -1);
+		if (optup == (Operator) NULL)
+			return node;
+
+		less_equal = make_op_with_func(oprid(optup), BOOLOID, false,
+										(Expr *) leftop, (Expr *) concat_expr,
+										InvalidOid, server_collation_oid, oprfuncid(optup));
+		constant_suffix = make_and_qual((Node *) greater_equal, (Node *) less_equal);
+		if (like_entry.is_not_match)
+		{
+			constant_suffix = (Node *) make_notclause((Expr *) constant_suffix);
+			ret = make_or_qual(node, constant_suffix);
+		}
+		else
+		{
+			constant_suffix = make_and_qual((Node *) greater_equal, (Node *) less_equal);
+			ret = make_and_qual(node, constant_suffix);
+		}
+		ReleaseSysCache(optup);
+	}
+	return ret;
+}
+
+/*
+ * Function responsible for obtaining unaccented version of input
+ * string with the help of ICU provided APIs. 
+ * We use a transformation rule to transliterate the string
+ */
+
+PG_FUNCTION_INFO_V1(remove_accents);
+Datum
+remove_accents(PG_FUNCTION_ARGS)
+{
+	text *arg1 = PG_GETARG_TEXT_PP(0);
```

Comment:
Why do we need to store arg1 in the first place?

Reply:
We do not need it, updated in latest rev